cmake/CodeAnalysis.cmake

```diff
@@ -0,0 +1,23 @@
+if (CMAKE_GENERATOR MATCHES "Visual Studio")
+
+    # Read checks from the clang tidy file using PowerShell
+    execute_process(COMMAND powershell "Select-String -path ${CMAKE_CURRENT_SOURCE_DIR}/.clang-tidy \"Checks:[\\s\\t]*[`\"'](.*)[`\"']\" -AllMatches  | Foreach-Object {$_.Matches} | Foreach-Object {$_.Groups[1].Value}" OUTPUT_VARIABLE ClangTidyChecks)
```

Comment:
VS seems to read the clang-tidy rules from the config file in the recent versions. So, this might not be needed anymore.

Reply:
https://github.com/lefticus/cpp_starter_project/pull/101/commits/60c0a5e43ed4dda43628b57cd040a04317ffff1b